python/tchannel/tornado/message_factory.py

```diff
@@ -154,11 +165,21 @@ def build_request(self, message, message_id=None):
 
         args = self.prepare_args(message)
 
+        tracing = Trace(
+            trace_id=message.tracing.trace_id,
+            span_id=message.tracing.span_id,
+            parent_span_id=message.tracing.parent_id,
+            endpoint=Endpoint(self.connection.remote_host,
+                              self.connection.remote_host_port,
```

Comment:
Again, given that you need just those two attributes, pass only those in instead of consuming `connection`.